exercises/src/main/scala/fpinscala/localeffects/LocalEffects.scala

```diff
@@ -93,12 +93,18 @@ sealed abstract class STArray[S,A](implicit manifest: Manifest[A]) {
   }
 
   // Read the value at the given index of the array
-  def read(i: Int): ST[S,A] = ST(value(i))
+  def read(i: Int): ST[S,A] = ST(value.applyOrElse(i, { (_: Int) =>
+    sys.error(s"Index out of bound ($i) for array of size ${value.length}")
+  }))
 
   // Turn the array into an immutable list
   def freeze: ST[S,List[A]] = ST(value.toList)
 
-  def fill(xs: Map[Int,A]): ST[S,Unit] = ???
+  def fill(xs: Map[Int,A]): ST[S,Unit] = {
+    xs.foldLeft(ST[S, Unit](())) { case (st, (index, a)) =>
```

Comment:
I guess it's personal preference, I find `({})` more confusing, since the normal way to represent unit is `()`, I had to test your version in a REPL to confirm it was actually the same.